index.js

```diff
@@ -185,7 +185,9 @@ const prepareOptions = state => {
 			if (userSignal.aborted) {
 				throw new DOMException('The operation was aborted.', 'AbortError')
 			}
-			userSignalHandler = arg => abortController.abort(arg)
+			userSignalHandler = () => {
+				abortController.abort(new Error('AbortError'))
+			}
 			userSignal.addEventListener('abort', userSignalHandler, {
 				once: true,
 			})
```

Comment:
>And the timeout would cause an Abort Error

I don't think so. New Undici throws the abort reason as-is, so now aborts caused by timeouts should immediately throw TimeoutError.

You can test it by pasting a following to REPL:

```javascript
const { fetch } = require("undici");

class CustomError extends Error {
	constructor(message) {
		super(message);
		this.name = "CustomError";
	}
}

let ac = new AbortController();
ac.abort(new CustomError("foo"));
await fetch("http://example.com", { signal: ac.signal }).catch(console.error);

ac = new AbortController();
ac.abort();
await fetch("http://example.com", { signal: ac.signal }).catch(console.error);
```

It also means that you can get rid of fragments like this, since we will get the actual error anyway with no need of detecting it:
```javascript
if (error.name === 'AbortError' && state[STATE_INTERNAL].timedout) {
	// eslint-disable-next-line no-ex-assign
	error = new TimeoutError(state[STATE_INTERNAL].timedout, state)
}
```

Reply:
Ok, my previous message is solved by upgrading undici to `5.23.0`